specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2024-09-01/DataConnectors.json

```diff
@@ -0,0 +1,1696 @@
+      "required": [
+        "tenantId",
+        "userId",
+        "publicFingerprint",
+        "pemFile"
+      ],
+      "x-ms-discriminator-value": "Oracle"
+    },
+    "SessionAuthModel": {
+      "description": "Model for API authentication with session cookie.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CcpAuthConfig"
+        }
+      ],
+      "properties": {
+        "userName": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The user name attribute key value."
+        },
+        "password": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The password attribute name."
+        },
+        "queryParameters": {
+          "type": "object",
+          "additionalProperties": {},
+          "description": "Query parameters to session service endpoint."
+        },
+        "isPostPayloadJson": {
+          "type": "boolean",
+          "x-nullable": true,
+          "description": "Indicating whether API key is set in HTTP POST payload."
+        },
+        "headers": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "HTTP request headers to session service endpoint."
+        },
+        "sessionTimeoutInMinutes": {
+          "type": "integer",
+          "format": "int32",
+          "x-nullable": true,
+          "description": "Session timeout in minutes."
+        },
+        "sessionIdName": {
+          "type": "string",
+          "description": "Session id attribute name from HTTP response header."
+        },
+        "sessionLoginRequestUri": {
+          "type": "string",
+          "description": "HTTP request URL to session service endpoint."
+        }
+      },
+      "type": "object",
+      "required": [
+        "userName",
+        "password"
+      ],
+      "x-ms-discriminator-value": "Session"
+    },
+    "CcpResponseConfig": {
+      "description": "A custom response configuration for a rule.",
+      "properties": {
+        "eventsJsonPaths": {
+          "description": "The json paths, '$' char is the json root.",
+          "example": "'$', '$.someProperty'",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "successStatusJsonPath": {
+          "description": "The value where the status message/code should appear in the response.",
+          "type": "string"
+        },
+        "successStatusValue": {
+          "description": "The the status value.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "isGzipCompressed": {
+          "description": "The value indicating whether the remote server support Gzip and we should expect Gzip response.",
+          "type": "boolean"
+        },
+        "compressionAlgo": {
```

Comment:
>compressionAlgo [](http://example.com/codeflow?start=9&length=15)

can you add some examples on what possible values look like?

Reply:
Update the description